mne/channels/tests/test_interpolation.py

```diff
@@ -52,6 +52,13 @@ def test_interpolation():
     # drops to 0.8
     thresh = 0.80
 
+    # check that interpolation does nothing if no bads are marked
+    epochs_eeg.info['bads'] = []
+    evoked_eeg = epochs_eeg.average()
+    for ch in evoked_eeg.info['chs']:
+        ch['loc'][:3] = np.array([0., 0., 0.])
```

Comment:
why do you need this step?

Reply:
Ah because you would get the error only if channel locations were not present. I tried for sample data, and if you do, `evoked.info['bads'] = []` it does not give any error

Reply:
Should not be necessary indeed

Reply:
I don't think you can reproduce the bug without this step

Reply:
@stralu can you post what error you see?

Reply:
I saw an:
> UnboundLocalError: local variable 'iprint' referenced before assignment

Turns out that the actual fix was to set the verbose level to false as suggested here  #5029.

Reply:
Still, I think its worth to fix the if statement

Reply:
or remove it

Reply:
So, I can't provide a better verification method

Reply:
I just tested this PR and test passes fine without hacking the channel locations.
I still don't get why you needed this.

Reply:
I think @jasmainak added this hacking because the test passes also when the if statement is broken - the function works nicely without the if statement, but needs more execution time. One way to verfify that the if works properly is to use a setting that leads to an error (no matter if a bad is set or not). This setting occurs in a function after the bespoken if and therefore it is possible to verify the ifs functioning.
By the way: I described the issue badly, so I updated it right now.

Reply:
For me, the test would make sense if it breaks on master and passes on the PR. This is what it was doing. Anyway, the tweak also LGTM